generator/testdata/google/example/library/v1/library_service_stub.gapic.cc.baseline

```diff
@@ -124,14 +126,208 @@ class DefaultLibraryServiceStub : public LibraryServiceStub {
     ::google::example::library::v1::GetBookRequest const& request,
     ::google::example::library::v1::Book* response) override {
     grpc::ClientContext grpc_ctx;
-    context->PrepareGrpcContext(&grpc_context);
+    context->PrepareGrpcContext(&grpc_ctx);
     return google::gax::GrpcStatusToGaxStatus(grpc_stub_->GetBigBook(&grpc_ctx, request, response));
   }
 
  private:
   std::unique_ptr<::google::example::library::v1::LibraryService::StubInterface> grpc_stub_;
 }; // DefaultLibraryServiceStub
 
+class RetryLibraryServiceStub : public LibraryServiceStub {
+ public:
+  RetryLibraryServiceStub(std::unique_ptr<LibraryServiceStub> stub_,
+                         gax::RetryPolicy const& retry_policy,
+                         gax::BackoffPolicy const& backoff_policy) :
+            next_stub_(std::move(stub)),
+            retry_policy_(retry_policy.clone()),
+            backoff_policy_(backoff_policy.clone()) {}
+
+  google::gax::Status
+  CreateBook(google::gax::CallContext& context,
+             ::google::example::library::v1::CreateBookRequest const& request,
+             ::google::example::library::v1::Book& response) {
+    auto retry_policy = (ctx.retry_policy_ ? ctx.retry_policy_ : retry_policy_)->clone();
+    auto backoff_policy = (ctx.backoff_policy_ ? ctx.backoff_policy_ : backoff_policy_)->clone();
+    gax::MethodInfo info = ctx.Info();
+    while (true) {
+      retry_policy->Setup(context);
+      gax::Status status = next_stub_->CreateBook(context, request, response);
```

Comment:
This will reuse the context object across calls, which is not valid. (What if a stub underneath this one adds metadata? Then that metadata will occur once in the first call, twice in the second call, etc.) We need to copy the context for each retry. (We can't even avoid a copy in the case that the first call is successful, because we don't know that it will succeed ahead of time)

Reply:
Okay, so we can't even reuse the CallContext. Custom copy ctor to the rescue.